workers/ohsome_quality_analyst/indicators/mapping_saturation/indicator.py

```diff
@@ -35,157 +35,115 @@ def __init__(
+        # (mixture of Gröchenig et al. +  Barrington-Leigh)
+        # saturation: 0 < f‘(x) <= 0.03 and years with saturation > 2
+        self.threshold_yellow = 0.03
+        # TODO: THRESHOLD_RED (where start stadium ends) with function from MA Katha p24
+        # self.threshold_red = None
 
         # Attributes needed for result determination
-        self.saturation = None
-        self.growth = None
+        self.best_fit: Optional[models.BaseStatModel] = None
+        self.fitted_models: List[models.BaseStatModel] = []
+        self.saturation: Optional[float] = None
+        self.growth: Optional[float] = None
 
     async def preprocess(self) -> None:
         query_results = await ohsome_client.query(
-            layer=self.layer, bpolys=self.feature.geometry, time=self.time_range
+            layer=self.layer,
+            bpolys=self.feature.geometry,
+            time=self.time_range,
         )
-        self.values = [item["value"] for item in query_results["result"]]
-        self.timestamps = [
-            isoparse(item["timestamp"]) for item in query_results["result"]
-        ]
-        # Latest timestamp of ohsome API results
-        self.result.timestamp_osm = self.timestamps[-1]
-        max_value = max(self.values)
-        if max_value == 0:
-            self.no_data = True
-        elif self.values[-1] == 0:
-            self.deleted_data = True
-        else:
-            self.values_normalized = [value / max_value for value in self.values]
+        for item in query_results["result"]:
+            self.values.append(item["value"])
+            self.timestamps.append(isoparse(item["timestamp"]))
 
     def calculate(self) -> None:
-        """Calculate the growth rate and saturation level within the last 3 years."""
-        if self.no_data:
-            self.result.description = "No features were mapped in this region."
+        # Latest timestamp of ohsome API results
+        self.result.timestamp_osm = self.timestamps[-1]
+        if not self.check_corner_cases():
             return
-        if self.deleted_data:
+        xdata = np.asarray(range(len(self.timestamps)))
+        for model in (
+            models.Sigmoid,
+            models.SSlogis,
+            models.SSdoubleS,
+            models.SSfpl,
+            models.SSasymp,
+            models.SSmicmen,
+        ):
+            logging.info("Run {}".format(model.name))
+            try:
+                self.fitted_models.append(
+                    model(xdata=xdata, ydata=np.asarray(self.values))
+                )
+            except RRuntimeError as error:
+                logging.warning(
+                    "Could not run model {0} due to RRuntimeError: {1}".format(
+                        model.name, error
+                    )
+                )
+                continue
+        self.best_fit = min(self.fitted_models, key=lambda m: m.mae)
+        logging.info("Best fitting model: " + self.best_fit.name)
+        self.saturation = (
```

Comment:
I think from here on we need to double check how we would like to approach this now. In the past we based our quality label on the saturation level. However, we could now change this and rather rely on the asymptote value of the fitted function.

If we decide to go for the asymptote approach we need to change the thresholds. The good thing would be that we can derive self.value as the completeness